packages/create-react-app/index.js

```diff
@@ -182,7 +182,25 @@ function run(root, appName, version, verbose, originalDirectory, template) {
 
   install(allDependencies, verbose, function(code, command, args) {
     if (code !== 0) {
-      console.error(chalk.cyan(command + ' ' + args.join(' ')) + ' failed');
+      console.log();
+      console.error('Aborting installation.', chalk.cyan(command + ' ' + args.join(' ')), 'has failed.');
+      // On 'exit' we will delete these files from target directory.
+      var knownGeneratedFiles = [
+        'package.json', 'npm-debug.log', 'yarn-error.log', 'node_modules'
```

Comment:
I thought about this, but I haven't been able to reproduce this behavior.
... and I assume it only happens for collision-avoidance. I don't think it'd ever happen during an install. + newer versions of npm write to a cache directory instead of the working directory.